src/Rings/mpoly-graded.jl

```diff
@@ -1453,7 +1453,7 @@ end
 #########################################
 function add_relshp(R, S, h)
   #this assumes that h is essentially a canonical map from R -> S
-  D = get_attribute!(() -> Dict{Any, Any}(), R, :relshp)::Dict{Any, Any}
+  D = get_attribute!(R, :relshp, Dict{Any, Any}())::Dict{Any, Any}
```

Comment:
The new code always allocates a new dictionary even if it is not needed. But you can do this:
```suggestion
  D = get_attribute!(Dict{Any, Any}, R, :relshp)::Dict{Any, Any}
```

Reply:
All tests just passed, this can be done later

Reply:
Oooohhhh, that's sooo smart 😅